frappe/utils/nestedset.py

```diff
@@ -59,15 +59,15 @@ def update_add_node(doc, parent, parent_field):
 
 	# get the last sibling of the parent
 	if parent:
-		left, right = frappe.db.sql("select lft, rgt from `tab{0}` where name=%s"
-			.format(doctype), parent)[0]
-		validate_loop(doc.doctype, doc.name, left, right)
+		whse_data = frappe.db.sql("select lft, rgt from `tab{0}` where name=%s"
+			.format(doctype), parent, as_dict=1)
+		if whse_data is not None and len(whse_data)!=0:
```

Comment:
Every parent must have lft and rgt. If the sql does not return any record, it means parent is invalid. May be we can add a validation saying "parent is invalid".

Reply:
Hi @nabinhait 
As per your suggestions, added validation. Please have look at the changes.